In [1]:
import numpy as np
import pandas as pd

In [2]:
#path = 'M:/My Documents/Business/Bugisiw Ltd/TaxReturn/TaxReturn_2020-2021/LLP accounts/'
taxreturn_yr='TaxReturn_2021-2022'
path = 'E:\\dtuklaptop\\e\\Users\\Mat\\python\\data\\property\\'+taxreturn_yr+'\\LLP accounts\\'
startdatestr='2021-04-06'
enddatestr='2022-04-05'

In [3]:
propertyidmap = {
2:'321LON',
1:'F1321LON',
14:'F2321LON',
3:'F3321LON',
4:'F4321LON',
7:'169FAW',
5:'F1169FAW',
6:'F2169FAW',
8:'F3169FAW',
73:'F1171FAW',
74:'F2171FAW',
75:'F3171FAW',
12:'163FRA',
9:'F2163FRA',
10:'F3163FRA',
11:'F4163FRA',
17:'SHOP196KIN',
15:'196AKIN',
44:'196AKIN',
16:'196BKIN',
19:'23BHAM',
34:'23CHAM',
18:'23CHAM',
31:'F58ALH',
30:'F68ALH',
33:'F78ALH',
32:'F88ALH',
26:'F1746ALH',
22:'F1846ALH',
21:'F1946ALH',
25:'F2046ALH',
28:'F2146ALH',
23:'F2246ALH',
27:'F2346ALH',
46:'F31214ALH',
52:'F41214ALH',
47:'F51214ALH',
48:'F71214ALH',
49:'F111214ALH',
53:'F101214ALH',
54:'F141214ALH',
55:'F161214ALH',
66:'F11618ALH',
67:'F31618ALH',
63:'F61618ALH',
68:'F71618ALH',
64:'F111618ALH',
71:'F121618ALH',
70:'F131618ALH',
65:'F161618ALH',
81:'RSA'
}

In [4]:
startdate = pd.to_datetime(startdatestr).date()
enddate = pd.to_datetime(enddatestr).date()

In [5]:
input_file_6045 = path + '6045_categorisedAndBealsReconciled.xlsx'
df6=pd.read_excel(input_file_6045,sheet_name='6045')
df6=df6[~df6.Account.isnull()]
df6=df6[(df6.Date >= startdate)&(df6.Date <= enddate)]

input_file_3072 = path + '3072_categorisedAndBealsReconciled.xlsx'
df3=pd.read_excel(input_file_3072,sheet_name='3072')
df3=df3[~df3.Account.isnull()]
df3=df3[(df3.Date >= startdate)&(df3.Date <= enddate)]

C:\Users\MSTDTP21\.conda\envs\mtpython36\lib\site-packages\ipykernel_launcher.py:4: FutureWarning: Comparing Series of datetimes with 'datetime.date'.  Currently, the
'datetime.date' is coerced to a datetime. In the future pandas will
not coerce, and a TypeError will be raised. To retain the current
behavior, convert the 'datetime.date' to a datetime with
'pd.Timestamp'.
  after removing the cwd from sys.path.
C:\Users\MSTDTP21\.conda\envs\mtpython36\lib\site-packages\ipykernel_launcher.py:9: FutureWarning: Comparing Series of datetimes with 'datetime.date'.  Currently, the
'datetime.date' is coerced to a datetime. In the future pandas will
not coerce, and a TypeError will be raised. To retain the current
behavior, convert the 'datetime.date' to a datetime with
'pd.Timestamp'.
  if __name__ == '__main__':


The following file is exported from Property App view INCOME_EXPENDITURE_TAX_LLP_ALL for this tax year.

Don't forget to add the header row.

In [6]:
input_file_beals = path + 'inc_exp_2022_240622_llp_all_19.xls'
dfR=pd.read_excel(input_file_beals,sheet_name='Sheet 1')
dfR=dfR[(dfR.date >= startdate)&(dfR.date <= enddate)]

C:\Users\MSTDTP21\.conda\envs\mtpython36\lib\site-packages\ipykernel_launcher.py:3: FutureWarning: Comparing Series of datetimes with 'datetime.date'.  Currently, the
'datetime.date' is coerced to a datetime. In the future pandas will
not coerce, and a TypeError will be raised. To retain the current
behavior, convert the 'datetime.date' to a datetime with
'pd.Timestamp'.
  This is separate from the ipykernel package so we can avoid doing imports until


In [7]:
for index, row in dfR.iterrows():
    dfR.at[index, 'Property']=propertyidmap[row.property_id]

### Check Bank data against Rent Statements
- differences in values indicate that some items may be included in Rent Statements that are missing from Bank
- INVESTIGATE EACH PROPERTY WHERE THE DIFFERENCE IS NOT ZERO - this is a data integrity check

In [8]:
dfBank=pd.concat([df6,df3]).sort_values(by=['Date'])
dfBG=dfBank.groupby('Property')['Amount'].sum()

#### Check all properties

In [9]:
dfRG=dfR.groupby('Property')['amount'].sum()

In [10]:
dfCheck=pd.concat([dfBG, dfRG], axis=1)
dfCheck.rename(columns={'Amount':'AmountB'},inplace=True)
dfCheck.rename(columns={'amount':'AmountRS'},inplace=True)
dfCheck['Diff']=round(dfCheck.AmountB - dfCheck.AmountRS,2)
dfCheck

AmountB  AmountRS    Diff
Property                             
169FAW     -1094.28  -1094.28    0.00
196AKIN     3606.39   3606.39   -0.00
196BKIN     6905.89   6905.89    0.00
23BHAM      8155.99   8155.99    0.00
23CHAM      3137.72   3137.72    0.00
321LON     -3569.07  -3569.07    0.00
F101214ALH  3253.39   3253.39    0.00
F111214ALH  4908.20   4908.20    0.00
F111618ALH  3756.56   3756.56    0.00
F11618ALH   4855.91   4855.91    0.00
F1169FAW    3462.22   3041.26  420.96
F1171FAW    -109.83   -109.83   -0.00
F121618ALH  3466.16   3466.16    0.00
F131618ALH  1755.87   1755.87    0.00
F1321LON    1440.72   1446.72   -6.00
F141214ALH   923.95    923.95    0.00
F161214ALH  3456.60   3456.60    0.00
F161618ALH  3659.63   3659.63   -0.00
F1746ALH    4290.36   4290.36    0.00
F1846ALH    5567.75   5567.75    0.00
F1946ALH    5414.72   5414.72    0.00
F2046ALH    5757.74   5757.74    0.00
F2146ALH    3288.54   3288.54    0.00
F2169FAW    4715.06   4715.06    0.00
F2171FAW    3178.69   3178.69    0.00
F2246ALH    5743.64   5743.64    0.00
F2321LON    5363.65   5363.65    0.00
F2346ALH    6066.26   6066.26    0.00
F31214ALH   3228.62   3234.62   -6.00
F31618ALH   4988.65   4988.65    0.00
F3163FRA    1528.24   1528.24   -0.00
F3169FAW    1713.05   1719.05   -6.00
F3171FAW    6827.68   6827.68    0.00
F3321LON    5196.95   5196.95    0.00
F41214ALH   3149.70   3137.70   12.00
F4163FRA    4518.48   4518.48    0.00
F4321LON   -1026.76  -1056.47   29.71
F51214ALH   2923.74   2923.74    0.00
F58ALH      5795.94   5795.94    0.00
F61618ALH   2975.22   2975.22    0.00
F68ALH      3447.03   3447.03    0.00
F71214ALH   2705.42   2705.42    0.00
F71618ALH   1555.71   1555.71    0.00
F78ALH      2950.03   2944.03    6.00
F88ALH      2116.01   2115.99    0.02
RSA        -1587.92  -1587.92    0.00
SHOP196KIN  3302.84   3005.64  297.20

F1 168 - received payment in bank of £420.96 on 10/02/2022 but not in Rent Statement data - could be over payment

F4 321 - £29.71 paid on 06/04/21 - shows in bank but not Rent Statement data

196 Shop - £297.20 are the mortgage payments taken from the Natwest account - note that the mortgage stopped in July 2021

#### Check individual properties (to pin point problems)

In [11]:
pty='F41214ALH'

In [12]:
dfBank.loc[dfBank.Property==pty].groupby(['Category'])['Amount'].sum()

Category
AgentFees            -720.00
BealsRentalIncome    2424.60
Deposit                 0.00
Maintenance         -3505.25
Mortgage             -649.65
RentalIncome         5600.00
Name: Amount, dtype: float64

In [13]:
dfR.loc[(dfR.Property==pty)].groupby(['cat'])['amount'].sum()

cat
Council             -12.30
Lettings Agency    -932.40
Maintenance       -3492.95
Mortgage           -649.65
Tenancy Related    8225.00
Name: amount, dtype: float64

In [14]:
dfR[(dfR.Property==pty)&(dfR.cat.str.contains('Lettings Agency|Tenancy Related'))].loc[dfR.Property==pty]['amount'].sum()

7292.599999999999

In [15]:
dfR[(dfR.Property==pty)&(dfR.cat.str.contains('Maintenance|Utilities'))].loc[dfR.Property==pty]['amount'].sum()

-3492.95

### Overall view of bank data

Total figures for both accounts followed by individual accounts.

In [ ]:
dfBank.groupby(['Category'])['Amount'].sum()

In [ ]:
df6.groupby(['Category'])['Amount'].sum()

In [ ]:
df3.groupby(['Category'])['Amount'].sum()

Check monthly P/L:

In [ ]:
input_file_6045 = path + '6045_categorisedAndBealsReconciled.xlsx'
df6=pd.read_excel(input_file_6045,sheet_name='6045')
df6=df6[~df6.Account.isnull()]
df6=df6[(df6.Date >= startdate)&(df6.Date <= enddate)]

input_file_3072 = path + '3072_categorisedAndBealsReconciled.xlsx'
df3=pd.read_excel(input_file_3072,sheet_name='3072')
df3=df3[~df3.Account.isnull()]
df3=df3[(df3.Date >= startdate)&(df3.Date <= enddate)]

#input_file_IV = path + '300 mvt .xlsx'
#dfIV=pd.read_excel(input_file_IV,sheet_name='data (36)')
#dfIV=dfIV[~dfIV.Account.isnull()]
#dfIV=dfIV[(dfIV.Date >= startdate)&(dfIV.Date <= enddate)]

#df6['Amount'][(df6.Date>='06/01/2020')&(df6.Date<='06/30/2020')].groupby(by=[df6.Date.dt.date]).sum()
#dfIV['Amount'][(dfIV.Date>='06/01/2020')&(dfIV.Date<='06/30/2020')].groupby(by=[dfIV.Date.dt.date]).sum()

In [ ]:
df6['Amount'].groupby(by=[df6.Date.dt.year, df6.Date.dt.month]).sum()

In [ ]:
df3['Amount'].groupby(by=[df3.Date.dt.year, df3.Date.dt.month]).sum()

In [ ]:
df6['Amount'].sum()

In [ ]:
df3['Amount'].sum()

### Split rent dates - may need to move this somewhere else
- this is used to identify rents that cross start/end periods of financial year
- probably should be put in journals generation

In [ ]:
import re
from datetime import datetime

start_period=datetime.strptime('06/04/2021', "%d/%m/%Y").date()
end_period=datetime.strptime('05/04/2022', "%d/%m/%Y").date()

for index, row in df6[(df6.Category=='RentalIncome')].iterrows():
    #print('>>>>>>>>>>>>>>>>>>>>>>>>>')
    search=re.search("^Rent for period ([0-9]{2}\/[0-9]{2}\/[0-9]{4})\s*-\s*([0-9]{2}\/[0-9]{2}\/[0-9]{4}) from.*$", row.Description)
    #print(search.group(1))
    #print(search.group(2))
    start_date_str=search.group(1)
    if(start_date_str=='30/02/2022'): start_date_str='01/03/2022' # Correct dates
    start_date=datetime.strptime(start_date_str, "%d/%m/%Y").date()
    end_date_str=search.group(2)
    if(end_date_str=='29/02/2022'): end_date_str='01/03/2022'  # Correct dates
    if(end_date_str=='30/02/2022'): end_date_str='01/03/2022'  # Correct dates
    end_date=datetime.strptime(end_date_str, "%d/%m/%Y").date()
    if((start_date<start_period) and (end_date>start_period) ):
        print(row)
    if((start_date<end_period) and (end_date>end_period) ):
        print(row)

        
        
# Identify rent period records
#df6[(df6.Category=='RentalIncome')&(df6.Description.str.contains("Rent for period ([0-9]{2}\/[0-9]{2}\/[0-9]{4}) - ([0-9]{2}\/[0-9]{2}\/[0-9]{4})"))]    